src/data/Dhis2Events.ts

```diff
@@ -1,13 +1,18 @@
+import _ from "lodash";
 import { D2Api } from "../types/d2-api";
 import { Event } from "../domain/entities/DhisDataPackage";
 import { SynchronizationResult } from "../domain/entities/SynchronizationResult";
 import { ImportPostResponse, postImport } from "./Dhis2Import";
 import i18n from "../locales";
+import { promiseMap } from "../utils/promises";
 
-export async function postEvents(api: D2Api, events: Event[]): Promise<SynchronizationResult> {
-    return postImport(() => api.post<ImportPostResponse>("/events", {}, { events }).getData(), {
-        title: i18n.t("Data values - Create/update"),
-        model: i18n.t("Event"),
-        splitStatsList: true,
+export async function postEvents(api: D2Api, events: Event[]): Promise<SynchronizationResult[]> {
```

Comment:
[nothing to do] Typically, we'd keep the result type as-is (singular), and do a merging on each iteration to accumulate the result. However, in this particular case, the parent caller is already working with array of results, so this approach is also fine.